app/api/helpers/files.py

```diff
@@ -196,6 +196,59 @@ def create_save_image_sizes(image_file, image_sizes_type, unique_identifier=None
     return new_images
 
 
+def create_system_image(image_file=None, upload_path=None, unique_identifier=None,
+                        ext='jpg', remove_after_upload=False):
+    """
+    Create System Images for Event Topics
+    :param upload_path:
+    :param ext:
+    :param remove_after_upload:
+    :param image_file:
+    :return:
+    """
+    # Get an unique identifier from uuid if not provided
+    filename = '{filename}.{ext}'.format(filename=get_file_name(), ext=ext)
+    if image_file:
+        data = urllib.request.urlopen(image_file).read()
+        image_file = io.BytesIO(data)
+    else:
+        file_relative_path = 'static/default_system_image.png'
+        image_file = app.config['BASE_DIR'] + '/' + file_relative_path
+    try:
+        im = Image.open(image_file)
+    except IOError:
```

Comment:
IOError handled here @srv-twry

Reply:
Really ?? 
```Python
except IOError:
    raise IOError(...)
```
How is this exception handling ?

Reply:
See the message `"Corrupt/Invalid Image"`

Reply:
That is due to the exception handling in the `if` condition. I am suggesting you to add it in the `else` block also just to be safe.

Reply:
@bhaveshAn Status on this please ?

Reply:
Solved this 2 days back. Check app.api.event_topics.py under Files Changed section of this PR.